src/dfx/src/lib/root_key.rs

```diff
@@ -1,11 +1,12 @@
 use crate::{lib::error::DfxResult, Environment};
-
+use anyhow::anyhow;
 use dfx_core::network::root_key;
 
 pub async fn fetch_root_key_if_needed(env: &dyn Environment) -> DfxResult {
     let agent = env.get_agent();
     let network = env.get_network_descriptor();
-    root_key::fetch_root_key_when_non_mainnet(agent, network).await?;
+    root_key::fetch_root_key_when_non_mainnet(agent, network).await
+        .map_err(|e| anyhow!("Failed to fetch the root key, did you run 'dfx start' to start the local replica?\n{}", e))?;
```

Comment:
Please take a look at DiagnosedError. It may provide even more readable output, by first displaying the error, then displaying the explanation, and finally displaying a suggested action. You'd add it here with something like `.context(DiagnosedError::new(...))`.